core/matrix/batch_dense.cpp

```diff
@@ -105,15 +107,22 @@ void BatchDense<ValueType>::apply_impl(const BatchLinOp* alpha,
                                        const BatchLinOp* beta,
                                        BatchLinOp* x) const
 {
-    // TODO: Remove this when non-uniform batching kernels have been
-    // implemented
     if (!this->get_size().stores_equal_sizes() ||
         !this->get_stride().stores_equal_strides()) {
         GKO_NOT_IMPLEMENTED;
     }
-    this->get_executor()->run(batch_dense::make_apply(
-        as<BatchDense<ValueType>>(alpha), this, as<BatchDense<ValueType>>(b),
-        as<BatchDense<ValueType>>(beta), as<BatchDense<ValueType>>(x)));
+    if (auto bid = dynamic_cast<const BatchIdentity<ValueType>*>(b)) {
+        if (auto xdense = dynamic_cast<BatchDense<ValueType>*>(x)) {
+            xdense->add_scale(alpha, this, beta);
```

Comment:
Technically we could use `convert_and_copy_to` if we wanted to be flexible, but I think this is fine for now.